### ISSUE TITLE
A repeating group inside a non repeating group is always empty on page refresh

### ISSUE THREAD

**Issue by lorang92** [Altinn/app-frontend-react#281]
## Describe the bug
See title.

## To Reproduce
1. Setup an app with a group (non repeating) that contains a repeating group
2. Add data to the repeating group
3. Refresh page
4. See that repeating group is empty (even though form data is correct from backend) . Data is rendered as expected if you click "Legg til ny"

## Expected behavior
Repeating group inside a non repeating group should show data on page refresh.

**Comment by bjosttveit** [comment]
This bug seems to be caused by the function `getRepeatingGroups` in `formLayout.ts` which is only used by `initRepeatingGroupsSaga` to initialize the repeating groups. This function filters out a lot of elements that are children of other groups, including repeating groups. The issue seems to be mitigated by this change:


![Image](https://user-images.githubusercontent.com/47412359/195855368-c9e96bab-ee4d-443e-864d-72092f544121.png)

This will cause it not to filter out repeating groups since it checks if an `edit` field is present. However, i do not really understand what the purpose of the filtering is and so am not comfortable with calling this a solution before I understand this logic better.

The cypress tests seem to be fine except that it causes the same behaviour as removing the leaveOpen-logic in #544. It causes the deleting of the top row not to open the next row which is fine since this test is changed in #544 anyway. I just do not understand why this happens 😅 

Anyone have an idea? @olemartinorg @Magnusrm

**Comment by olemartinorg** [comment]
The `edit` property on a `Group` is only used for specifying some extra properties relevant for editing. It's not what defines a `Group` to be a repeating group.. :wink: A group is a repeating group if `element.type === 'Group' && element.maxCount > 1` (but the `edit` property is optional). I think you're on the right track apart from that!

**Comment by bjosttveit** [comment]
I agree on that. Checking the maxCount property is probably better. But I still wonder if this will have other side-effects.

**Comment by olemartinorg** [comment]
We can go through the code together later if you get something working, and evaluate any side-effects then. The `edit` property is just not part of what makes a `Group` a repeating one. So a `Group` with `maxCount <= 1` is not a **repeating group** and should probably be ignored when checked for in a function called `getRepeatingGroups`.

**Comment by olemartinorg** [comment]
This should not be an issue any longer in v4. Closing as completed.
